File: 프로그래머스/unrated/154538. 숫자 변환하기/숫자 변환하기.java
import java.util.Queue;
import java.util.LinkedList;

class Info{
    int y;
    int count;
    public Info(int y, int cnt){
        this.y = y;
        count = cnt;
    }
}
class Solution {
    public int solution(int x, int y, int n) {
        Queue<Info> que = new LinkedList<>();
        int answer = -1;

        que.add(new Info(y, 0));
        while(!que.isEmpty()){
            Info get = que.poll();
            if(get.y == x){
                answer = get.count;
                break;
            }

            if(get.y-n >= x) que.add(new Info(get.y-n, get.count+1));
            if(get.y/2 >= x && get.y%2 ==0) que.add(new Info(get.y/2, get.count+1));
            if(get.y/3 >= x && get.y%3 ==0) que.add(new Info(get.y/3, get.count+1));
        }
        return answer;
    }
}

File: 프로그래머스/unrated/154538. 숫자 변환하기/숫자 변환하기.js
function solution(x, y, n) {
    if(x === y) return 0;
    var answer = 0;
    var queue = [y];
    while (queue.length) {
        var len = queue.length
        console.log(queue);
        for(var i=0; i<len; i++){
            var num = queue.shift()
            if(num-n === x || num/2 === x || num/3 === x) return answer+1;
            
            if(num%3==0 && num/3 > x) queue.push(num/3);
            if(num%2==0 && num/2 > x) queue.push(num/2);
            if(num-n > x) queue.push(num-n);
        }
        answer++;
    }
    return -1;
}